### PR TITLE
hoptoad gem is replaced with airbrake gem

### LINKED ISSUE #1891: Replace the hoptoad_notifier gem with airbrake

### PR DESCRIPTION
**Pull request by namangupta01**
Closes #1891

### CODE REVIEW

File: config/initializers/hoptoad.rb [hunk 1]
```diff
@@ -1,29 +0,0 @@
...
-  else
-    config.development_environments = 'development test'
-  end
-
-  config.ignore_only = %w{
-    ActiveRecord::RecordNotFound
-    ActionController::InvalidAuthenticityToken
-    ActionController::UnknownAction
-    AbstractController::ActionNotFound
-    ActionView::MissingTemplate
```
Comment (AEtherC0r3) [inline config/initializers/hoptoad.rb]: You forgot to transfer this to the new config file

File: config/initializers/airbrake.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  end
+end
+
+Airbrake.add_filter do |notice|
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActiveRecord::RecordNotFound' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionController::InvalidAuthenticityToken' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionController::UnknownAction' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'AbstractController::ActionNotFound' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionController::UnknownFormat' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionController::RoutingError' }
```
Comment (AEtherC0r3) [inline config/initializers/airbrake.rb]: In the old config, it does some pattern matching before deciding whether on not to keep the exception

File: config/initializers/airbrake.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+end
+
+Airbrake.add_filter do |notice|
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActiveRecord::RecordNotFound' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionController::InvalidAuthenticityToken' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionController::UnknownAction' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'AbstractController::ActionNotFound' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionView::MissingTemplate' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionController::UnknownFormat' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionController::RoutingError' && exception_data[:error_message] =~ %r{\[GET\]} }
```
Comment (AEtherC0r3) [inline config/initializers/airbrake.rb]: It seems to me that exception_data isn't defined here

**Comment by Ana06** [comment]
@namangupta01 what is the state of this?

**Comment by namangupta01** [comment]
@AEtherC0r3 @Ana06 Please have a look!

File: config/initializers/hoptoad.rb (hunk 1 above)
Comment (bear454) [reply config/initializers/hoptoad.rb]: @AEtherC0r3 they are transferred in the current changeset... please update your review.

File: config/initializers/airbrake.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+end
+
+Airbrake.add_filter do |notice|
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActiveRecord::RecordNotFound' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionController::InvalidAuthenticityToken' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionController::UnknownAction' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'AbstractController::ActionNotFound' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionView::MissingTemplate' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionController::UnknownFormat' }
+  notice.ignore! if notice[:errors].any? { |error| error[:type] == 'ActionController::RoutingError' && error[:error_message] =~ %r{\[GET\]} }
```
Comment (AEtherC0r3) [inline config/initializers/airbrake.rb]: The `error` hash doesn't have an `:error_message` key (https://github.com/airbrake/airbrake-ruby/blob/master/lib/airbrake-ruby/nested_exception.rb)

**Comment by differentreality** [comment]
@namangupta01

**Comment by namangupta01** [comment]
PR updated!

**Comment by bear454** [comment]
@namangupta01 can you rebase on master & push again, to clear Harkiri? I'd like to get this merged in today!

**Comment by namangupta01** [comment]
@bear454 sure!

**Comment by namangupta01** [comment]
@bear454 done!

**Pull request merged**